pkg/extensions/search/gql_proxy/handler_utils.go

```diff
@@ -0,0 +1,69 @@
+package gqlproxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"maps"
+	"net/http"
+)
+
+func prepareAndWriteResponse(data map[string]any, response http.ResponseWriter) {
+	response.Header().Set("Content-Type", "application/json")
+
+	responseBody, err := json.MarshalIndent(data, "", "    ")
+	if err != nil {
+		http.Error(response, "failed to marshal response data", http.StatusInternalServerError)
+
+		return
+	}
+
+	_, err = response.Write(responseBody)
+	if err != nil {
+		http.Error(response, "failed to write response", http.StatusInternalServerError)
+
+		return
+	}
+}
+
+func deepMergeMaps(a, b map[string]any) map[string]any {
```

Comment:
@rchincha this is the new approach for aggregating the data. Since the response is a JSON, we can aggregate the data as a map type with individual logic for the embedded types - nested map, numeric types, and arrays.

This should be common across all handlers and may change when pagination comes into picture.